js/scorm/wrapper.js

```diff
@@ -1,787 +1,776 @@
-       */
-      this.lastCommitSuccessTime = null;
-      /**
-       * The exit state to use when course isn't completed yet
-       */
-      this.exitStateIfIncomplete = 'auto';
-      /**
-       * The exit state to use when the course has been completed/passed
-       */
-      this.exitStateIfComplete = 'auto';
-
-      this.timedCommitIntervalID = null;
-      this.retryCommitTimeoutID = null;
-      this.logOutputWin = null;
-      this.startTime = null;
-      this.endTime = null;
-
-      this.lmsConnected = false;
-      this.finishCalled = false;
-
-      this.logger = Logger.getInstance();
-      this.scorm = pipwerks.SCORM;
-      /**
-       * Prevent the Pipwerks SCORM API wrapper's handling of the exit status
-       */
-      this.scorm.handleExitMode = false;
-
-      this.suppressErrors = false;
-      this.debouncedCommit = _.debounce(this.commit.bind(this), 100);
-      if (window.__debug) {
-        this.showDebugWindow();
-      }
+    /**
+     * prevents commit from being called if there's already a 'commit retry' pending.
+     */
+    this.commitRetryPending = false;
+    /**
+     * how many times we've done a 'commit retry'
+     */
+    this.commitRetries = 0;
+    /**
+     * not currently used - but you could include in an error message to show when data was last saved
+     */
+    this.lastCommitSuccessTime = null;
+    /**
+     * The exit state to use when course isn't completed yet
+     */
+    this.exitStateIfIncomplete = 'auto';
+    /**
+     * The exit state to use when the course has been completed/passed
+     */
+    this.exitStateIfComplete = 'auto';
 
-      if ((window.API && window.API.__offlineAPIWrapper) || (window.API_1484_11 && window.API_1484_11.__offlineAPIWrapper)) {
-        this.logger.error('Offline SCORM API is being used. No data will be reported to the LMS!');
-      }
-    }
+    this.timedCommitIntervalID = null;
+    this.retryCommitTimeoutID = null;
+    this.logOutputWin = null;
+    this.startTime = null;
+    this.endTime = null;
 
-    // ******************************* public methods *******************************
+    this.lmsConnected = false;
+    this.finishCalled = false;
 
-    static getInstance() {
-      if (ScormWrapper.instance === null) {
-        ScormWrapper.instance = new ScormWrapper();
-      }
+    this.logger = Logger.getInstance();
+    this.scorm = pipwerks.SCORM;
+    /**
+     * Prevent the Pipwerks SCORM API wrapper's handling of the exit status
+     */
+    this.scorm.handleExitMode = false;
 
-      return ScormWrapper.instance;
+    this.suppressErrors = false;
+    this.debouncedCommit = _.debounce(this.commit.bind(this), 100);
+    if (window.__debug) {
+      this.showDebugWindow();
     }
 
-    getVersion() {
-      return this.scorm.version;
-    }
+    if (!(window.API?.__offlineAPIWrapper && window?.API_1484_11.__offlineAPIWrapper)) return;
```

Comment:
I think this should be
`if (!(window.API?.__offlineAPIWrapper && window?.API_1484_11?.__offlineAPIWrapper)) return;`